homeassistant/components/mqtt/light/schema_basic.py

```diff
@@ -298,13 +307,8 @@ def brightness_received(msg):
             self._brightness = percent_bright * 255
             self.async_write_ha_state()
 
-        if self._topic[CONF_BRIGHTNESS_STATE_TOPIC] is not None:
-            topics[CONF_BRIGHTNESS_STATE_TOPIC] = {
-                "topic": self._topic[CONF_BRIGHTNESS_STATE_TOPIC],
-                "msg_callback": brightness_received,
-                "qos": self._config[CONF_QOS],
-            }
-        elif (
+        add_topic(CONF_BRIGHTNESS_STATE_TOPIC, brightness_received)
+        if (
```

Comment:
The `elif` was not correct, we should restore the state if `self._optimistic_brightness` is `True`